src/ParseFile.js

```diff
@@ -418,39 +418,45 @@ const DefaultController = {
     if (source.format !== 'file') {
       throw new Error('saveFile can only be used with File-type sources.');
     }
-    // To directly upload a File, we use a REST-style AJAX request
-    const headers = {
-      'X-Parse-Application-ID': CoreManager.get('APPLICATION_ID'),
-      'Content-Type': source.type || (source.file ? source.file.type : null)
-    };
-    const jsKey = CoreManager.get('JAVASCRIPT_KEY');
-    if (jsKey) {
-      headers['X-Parse-JavaScript-Key'] = jsKey;
-    }
-    let sessionToken = options.sessionToken;
-    const userController = CoreManager.getUserController();
-    if (!sessionToken && userController) {
-      const currentUser = await userController.currentUserAsync();
-      sessionToken =  currentUser ? currentUser.getSessionToken() : undefined;
-    }
-    if (sessionToken) {
-      headers['X-Parse-Session-Token'] = sessionToken;
-    }
-    let url = CoreManager.get('SERVER_URL');
-    if (url[url.length - 1] !== '/') {
-      url += '/';
+    if (source.base64) {
```

Comment:
Is there a case where this would happen?

Reply:
Yes... when a new `Parse.File` is created with byte array it is converted to base64. Also when you call `save()` on a `Parse.File` that was created with a uri, it will download it to base64 before calling `saveFile()` in the controller

Reply:
In that case wouldn't `source.format` be `base64` instead of `file`